runbotics/runbotics-scheduler/src/websocket/bot/bot.gateway.ts

```diff
@@ -163,6 +168,25 @@ export class BotWebSocketGateway implements OnGatewayDisconnect, OnGatewayConnec
         this.logger.log(`<= Success: logs from bot ${installationId} saved`);
     }
 
+    @UseGuards(WsBotJwtGuard)
+    @SubscribeMessage(BotWsMessage.KEEP_ALIVE)
+    async processKeepAliveEventListener(
+        @ConnectedSocket() socket: BotAuthSocket,
+    ) {
+        const result = await this.botService.findById(socket.bot.id)
+            .then(bot => {
+                if (bot.status === BotStatus.DISCONNECTED)
+                    this.botService.save({ ...bot, status: BotStatus.CONNECTED });
+
+                return HttpStatus.OK;
+            })
+            .catch(error => {
+                return HttpStatus.INTERNAL_SERVER_ERROR;
+            });
+
+        return result;
+    }
+
```

Comment:
I thought more like creating a custom method updating the bot entity with a single query

bot doesn't care about response status it's more like `If you do something with it or not it's your business`